modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/upgrade/v1_0_1/SynonymsConfigurationUpgradeProcess.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.search.tuning.synonyms.web.internal.upgrade.v1_0_1;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.io.unsync.UnsyncByteArrayInputStream;
+import com.liferay.portal.kernel.io.unsync.UnsyncByteArrayOutputStream;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.List;
+
+import org.apache.felix.cm.file.ConfigurationHandler;
+
+/**
+ * @author Felipe Lorenz
+ */
+public class SynonymsConfigurationUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (!hasTable("Configuration_")) {
+			return;
+		}
+
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+				"select * from Configuration_ where configurationId = ?");
+			PreparedStatement preparedStatement2 = connection.prepareStatement(
+				"update Configuration_ set dictionary = ? where " +
+					"configurationId = ?")) {
+
+			preparedStatement1.setString(1, _CONFIGURATION_ID);
+
+			ResultSet resultSet = preparedStatement1.executeQuery();
+
+			if (resultSet.next()) {
+				String dictionaryString = resultSet.getString("dictionary");
+
+				if (Validator.isNull(dictionaryString)) {
+					return;
+				}
+
+				Dictionary<String, Object> dictionary =
+					ConfigurationHandler.read(
+						new UnsyncByteArrayInputStream(
+							dictionaryString.getBytes(StringPool.UTF8)));
+
+				List<String> synonymFilterNames = new ArrayList<>(
+					Arrays.asList((String[])dictionary.get("filterNames")));
+
+				synonymFilterNames.addAll(
```

Comment:
It seems like if someone already defined a filter named one of the _FILTER_NAMES, then it would be duplicated - is this correct? If so, then we would want to make sure it's not duplicated - and add logic to the test so that one of the filters is already predefined, and to make sure that it isn't duplicated by the upgrade.